File: cpp/sum-of-multiples/sum_of_multiples.cpp
#include "sum_of_multiples.h"

namespace sum_of_multiples {
	int to(std::vector<int> factors, int n)
	{
		std::set<int> multiples;

		for (int factor : factors) {
			for (int i = factor; i < n; i += factor)
				multiples.insert(i);
		}

		return std::accumulate(multiples.begin(), multiples.end(), 0);
	}
}  // namespace sum_of_multiples

File: csharp/sum-of-multiples/SumOfMultiples.cs
using System.Collections.Generic;
using System.Linq;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> factors, int max)
    {
        var multiples = new HashSet<int>();
        foreach (int factor in factors)
        {
            if (factor == 0)
                continue;

            for (int i = factor; i < max; i += factor)
                multiples.Add(i);
        }

        return multiples.Sum();

    }
}